### PR TITLE
Firing Pin System - The Lockbox Solution

### PR DESCRIPTION
**Pull request by Cheridan**
Adds Firing Pins system for guns.

Firing pins are items that are inserted into guns. Most guns start with a default firing pin, which runs no extra checks.  

Intended to solve 2 large problems:

A. Replace the Lockboxes in R&D. Firing pins are items that are inserted into guns. 
Most guns start with a default firing pin, which runs no extra checks. 
Guns printed in R&D start without firing pins. 

Adds 2 firing pin designs to R&D:
-Testing-range pin which allows a gun to fire in proximity to the testing range.
~~-High-grade Loyalty pin which allows a gun to fire if the user is loyalty implanted.~~

B. Fix issue of Nuke Ops not wanting to buy heavy weapons for fear that they will get turned against them.
Heavier op weapons such as l6 SAW, Bullpup, etc have syndicate pins which only fire if the user has a syndicate weapon auth. implant.

Nuke Ops now start with weapon auth. implants instead of Explosive Implants. Explosive implants were added to prevent crew getting Op equipment, but it mostly failed at this as you could still strip an unconscious operative and you'd always get his gun since it wasn't equipped onto him.
They also destroy all hope of RP with the ops. In a poll that was ran, most people voted that they would attempt to capture Ops alive as a matter of prestige and to allow RP such as interrogations.

Explosive implants are now purchasable. They now have the same explosion as a blobbed welding tank -- It's smaller than the 11-crystal syndibomb but, of course it requires you to die for it to go off.

### CODE REVIEW

**Comment by Cheridan** [comment]
TODO: Put a box or two of standard firing pins into the armory.

**Comment by hornygranny** [comment]
absolutely based

**Comment by pudl** [comment]
i hate this idea

**Comment by paprka** [comment]
Are you forgetting that ops frequently lose shit like agent IDs and belts and armor which could be turned against them really easily

I find minibombs in stolen military belts from ops who don't explode in time and use them to gib ops all the time, for example, now this would be 100x easier.

All this does is make looting ops easier, if anything, now that they don't have a default panic switch for taking out their gear if caught. It won't open up RP opportunity, it'll just make the shitler powergamer vigilantes murder AS FAST AS POSSIBLE and drag the body away so they can strip it and steal the gear and hog the phat lootz and op killer trophies.

Don't remove explosive implants. Ops are not a roleplay gamemode. You're not going to get roleplay from nuclear emergency just like you're not going to get roleplay from wizards 99% of the time. People may say they take ops captive but in the heat of it they'll just murder/strip/call shuttle to deny op greentext.

The rest of this PR owns though, lockboxes = shit

**Comment by tkdrg** [comment]
This is an amazing idea as far as the pins go, but the huge nerf to nuke ops is awful. Traitors should be able to buy something to override those pins, however, maybe an implant or even a nuke-style syndicate pin.

**Comment by optimumtact** [comment]
I like this PR because paprika doesn't

:+1:

**Comment by Cheridan** [comment]
I dunno. It's something to consider, but I think fluke ops manage to get looted enough even with default explosive implants. The guns are the biggest thing, though I suppose minibombs are powerful too (maybe I could add an implant check there too?)

Keep in mind that this also opens up the possibility of freeing arrested operative comrades from the brig. It's not a nerf or a buff, it's just a change.

**Comment by tkdrg** [comment]
It IS a nerf. You're making the guns even more powerful, and the more interesting tools (specially the stealthy ones) a lot less useful. Creative nuke tactics are shoved to the curb in exchange for run and gun which is very boring and not interesting at all. Furthermore, the single most valuable item the crew can loot from a nuke ops is the pinpointer, which is not locked.

**Comment by optimumtact** [comment]
They can loot it from them with the explosive implant as well. The way you're presenting it, explosive implants are somehow a magic solution to prevent looting, but that isn't true at all.

**Comment by tkdrg** [comment]
That's what they are, though. All semi-competent nuke ops (most of them, usually 3-4 out of the team of five from my experience) are going to blow themselves up before being looted. Now, if they want to do so, they'll have to either burn 9TC on it (which is a lot), or just buy a bunch of guns and forget about it. I really don't think we should be moving nuke ops in that direction, when the general consensus among coders seems to be encouraging less antag murderboner and more creative antagging.

**Comment by TZK13** [comment]
So testing this out the first thing I'll say is that all guns spawned via the game panel should come equipped with a firing pin.

Besides that I'll say that traitors working R&D should still be able to get guns as they please with something they can get via their uplink with the most obvious choice being an emag used on a gun removing all associated firing pin restrictions.

**Comment by TZK13** [comment]
Also unless I missed it I don't see any way to remove firing pins? You should really be able to if for example a scientist wants to test out a gun with a firing pin and then gets the clearance to be loyalty implanted for use of a loyalty pinned gun. Would also provide a way for syndie weapons to be still be used by the crew but make it require more effort to be done what with having to remove the pin and then finding a replacement pin to put in.

**Comment by Fox-McCloud** [comment]
Give ops dexplosive implants; they can pretty much call *deathgasp whenever they want to prevent people from looting--additionally, if they're in crit, they can just "succumb" and they'll die and immediately blow up.

**Comment by Delimusca** [comment]
this is fantastic! :+1: 
i forsee scisec walking in to the lab and leaving with 7 working AEGs though.

**Comment by paprka** [comment]
Just add the pins to nuke ops, no need to remove the explosive implant. They won't suicide if they're captured alive and want to roleplay. If they blow themselves up once captured? They weren't planning on roleplaying anyway.

**Comment by leibniztg** [comment]
Removing default explosive implants from nuke ops. is terrible. 
Guns arent the only critical thing that get looted, pinpointers are really important and they can make or break the round for the ops. And plenty of other useful items that you dont want in the hand of the enemy because your teammate was retarded and beelined for sec and got looted in 15 seconds.
I cant imagine teams wasting effort on freeing captured ops, since they should work on getting the disk and they were useless enough to get captured alive in the first place.

"Explosive implants were added to prevent crew getting Op equipment, but it mostly failed at this as you could still strip an unconscious operative and you'd always get his gun since it wasn't equipped onto him."
Usually nuke ops are unconscious because they are in crit, in which case they can succumb when they see people next to them looting them. Like they are supposed to.

Also, it is indeed a nerf to traitors who could emag open lockboxes. Probably something should be given to them in exchange. 
Another thing, with this traitors cannot use guns looted from sec with loyalty pins (would they have that? or would adding them to every gun every round be a new part of the powergaming routine?) which is also a huge unmentioned nerf to antagonists in general.
I think this "High-grade Loyalty pin" has potentially huge consequences for the game and we should consider them more carefully.

In general, I think this change has some okay ideas but it would make the game worse in its current form.
Some of the concepts in it should be given more thought before adding them.

**Comment by ike709** [comment]
How do you remove the firing pin from the gun?

**Comment by AlexCrimson** [comment]
Its a really cool idea, but its also a slippery slope. Hopefully we do not end up with id-locked guns screwing over traitors or non-antag hijinks.

**Comment by Iamgoofball** [comment]
oh god i was just thinking the other day 'what if someone codes MGS4 implant locked guns'

i kind of think this should be tied to the ID instead

**Comment by paprka** [comment]
BEHOLD

GUNS OF THE GUNCARGO

**Comment by Steelpoint** [comment]
Many Ops don't know that if you succumb while in crit that you will still blow up. That's why Ops often get looted.

I like the changes, but I still think Ops need their explosive implant.

**Comment by Perakp** [comment]
I guess it can work, but it certainly shifts the power balance between cargo and R&D, and crew and antagonist.

**Comment by Niknakflak** [comment]
Interesting Idea.  I would actually like to give this a test run for awhile to see it in action before I make some judgments.  Though, I will probably always buy an explosive implant anyway as a nuke op because I fucking love those things.  One of the reasons I love the implant bundle.  NOTHING is better than exploding when caught!

**Comment by Cheridan** [comment]
So I've heard some feedback on antags obtaining pins for their treachery. I'll look into adding some solution for this -- Firing pins for 1tc, or printable in R&D with Syndi research levels, etc. 

There currently is no way to remove or replace a firing pin. This will probably be added eventually, but it has to be something that's relatively difficult to do otherwise you Sci will be hoarding all the laser tag equipment.

**Comment by Miauw62** [comment]
Yeah, I agree that explosive implants should stay and if they go, they
should at least be cheaper.
Op 10-jan.-2015 07:56 schreef "Cheridan" <notifications@github.com>:

**Comment by Dorsisdwarf** [comment]
Traitors should be able to order a "Universal Activator" implant that lets them use any gun, irregardless of firing pins

How does this mesh with Summon Guns, since that spawns L6 SAWs?

File: code/modules/research/designs/weapon_designs.dm [hunk 1]
```diff
@@ -2,6 +2,26 @@
 /////////////////Weapons/////////////////
 /////////////////////////////////////////
 
+/datum/design/nuclear_gun/pin_testing
```
Comment (RemieRichards) [inline code/modules/research/designs/weapon_designs.dm]: what's with these funky paths? why is a pin a subtype of the nuclear_gun design datum?

**Comment by Iamgoofball** [comment]
Uh yeah @Cheridan make sure summon guns guns are pinless

Double check the wands and staves too

**Comment by Miauw62** [comment]
if theyre pinless they wont fire goofball.

**Comment by Lumipharon** [comment]
Don't put syndie pins in the low end shit. It's the ops fault for losing their guns, and with the stun changes it's not even as significant (ops have good bullet resist). That said, the lmg IS an issue in that if you make the huge investment in it, you also need to invest in an e-shield. I still would never have someone with a SAW without a shield, but atleast when they get tased it's less ggnore.

I'd prefer syndie pins being purchasable. If you think you're unrobust/getting an expensive gun like the SAW, a 1tc investment to lock it seems reasonable enough.

Is there a way to see what kind of pin a weapon has in it? Also there really needs to be a way to remove/replace pins (without turning lasertag weapons into prized loot).

**Comment by Iamgoofball** [comment]
then make sure all the summon X weapons have "anyone can use them pins"

**Comment by TZK13** [comment]
That's already been in since the beginning goofball.

**Comment by Cheridan** [comment]
@Lumipharon For now, I'll add a line on Examine that displays the pin type.

**Comment by Miauw62** [comment]
Make the laser tag pins only fire if you're wearing laser tag armor.
Op 11-jan.-2015 07:25 schreef "Cheridan" <notifications@github.com>:

**Comment by paprka** [comment]
^ yeah this

**Comment by RemieRichards** [comment]
@Miauw62 @paprka are you two blind? that's been in there since the PR was first put up.

**Comment by Razharas** [comment]
Make pins removable only if you can fire anyway

**Comment by Jordie0608** [comment]
Changelog :^)

**Comment by Jalleo** [comment]
@Cheridan will this require a special spawn menu for guns for admins to spawn guns with specific types of pins or none at all since the current item spawn menu does not support this?

**Comment by Delimusca** [comment]
is it be possible to have the protolathe strip pins? if admin/etc spawned science guns are indeed pinless, it would sort that problem.

**Comment by Jalleo** [comment]
@Delimusca the pins would probably be deconstructed with the gun as well. Cargo and R&D may be able to make more Firing Pins depending on what they are and will probably be secured in such a way to prevent mass abuse.
We just have to wait for Cheridan to respond to see what solutions are taken on

**Comment by dannno** [comment]
I keep trying to look at this from every angle but it's just such a MASSIVE nerf to antags and non antags and anyone not-security alike that it doesn't seem worth it

**Comment by Cheridan** [comment]
no it's not

**Comment by dannno** [comment]
What is it then? If it prevents do-gooders or traitors from using weapons it's a nerf to them, or am I really missing something here (yeah I realize it wouldn't prevent them "all of the time")

**Comment by Cheridan** [comment]
yes you're missing something here

**Comment by TZK13** [comment]
>Firing pins are items that are inserted into guns. Most guns start with a default firing pin, which runs no extra checks. 

That includes all of the guns security's armed with.

This IS a nerf to traitor scientists working R&D though as it stands currently.

**Comment by tkdrg** [comment]
Cheridan, please. It's bad enough that you want to encourage nuke ops murderboning and make emags less useful, but don't try to pretend that it isn't a major nerf to both of those, because it's glaringly obvious.

**Comment by Lumipharon** [comment]
I think the idea of having locking mechanisms for guns in the game is a good idea, but I don't like this implementation.
It hurts antags as it's harder for them to get weapons, while the whole 'non antags arming up' isnt really effected at all. Non antags need armory access currently, which only three people have. If the HoP is giving out armory access without proper authority he's set for demotion and brigging. With this change, either they need to get implanted, or if you can get standard pins from cargo, then it's back to guncargo.
As I've said before, syndie locked pins for 1tc would be good for ops, as it's an obvious choice for expensive shit like the SAW, but usually isn't worth it for the lesser guns like the shotgun, keeps things interesting.

Also what's wrong with lockboxes? They're actually secure unlike locked crates, and protolathes generating boxes is far from the most unrealistic thing in the game. Also if nothing else, it means you leave behind evidence, that can show whether it has been opened with access or emagged (unless you acid the empty boxes).

**Comment by Steelpoint** [comment]
Everyone here is retarded. Allow me to explain this.

ALL 
WEAPONS
AT
ROUNDSTART
HAVE
A
FIRING
PIN

THIS INCLUDES ARMOURY WEAPONS.

THE ONLY THING THIS PR CHANGES IS THAT ANY WEAPON THAT SCIENCE PRODUCES SIMPLY COMES WITHOUT A FIRING PIN AND NOT IN A BOX

PLEASE PEOPLE.

**Comment by Cheridan** [comment]
Pins are now emaggable to override them.
Adds standard pins to armory on the map.
Adds pin supply pack to cargo for lack of better means of replenishment.
You can examine a gun to see what firing pin is installed.

Loyalty pins raised to combat 10 and are unable to be constructed for now.

This is ready now.

**Comment by Lumipharon** [comment]
Are only the pins emaggable, or can you emag them while in the gun?

**Comment by Cheridan** [comment]
Just the pins for now. They'll be removable later and I didn't want to write redundant code.

**Comment by paprka** [comment]
Rename the pins to 'electronic firing pin' please, since you can emag it and all that

**Comment by Jalleo** [comment]
:+1:

**Comment by tkdrg** [comment]
Still going to ruin the balance of nuke ops, still :-1: from me. Buffing the L6 SAW of all things is awful, and removing the explosive implant encourages ops to focus on guns/mechs/syndiborgs/viscerators instead of the other more interesting items, which is terrible. I have no idea why you're so adamant about downplaying this, but it really is a big change in the direction of nuke, and not a good one.

**Comment by paprka** [comment]
We could really easily make the l6 a 5.56 LMG and do 35 damage like the carbine

Actually someone should probably do that and lower the price a bit.

**Comment by Cheridan** [comment]
@tkdrg I think the change is a good one. They are supposed to buy those big-ticket items in high-population rounds. That is why they are there, to provide extra firepower to counter the increased enemy force. 

What "other more interesting items" are you talking about? Chameleon projectors?

**Comment by tkdrg** [comment]
I've thought about it a bit more, and I suppose you're right. The 3 important items that this nerfs are syndie minibombs, chemsprayers and pinpointers. The first one was always BS so a nerf is fine, and the middle one was already due for a rebalancing (which I might PR soon[tm]), which leaves pinpointers as high-value loot. I'm not sure if that is going to make the ops too weak, but I'm willing to wait and see.
TL;DR: :+1: but pinpointers might need a buff after this

**Comment by Cheridan** [comment]
I'd pin chemsprayers if I could, but they're not technically guns. There's already a safeguard to prevent sprayers from being used against Ops though, which is that blood-red hardsuits have 100 bio protection.

I'll concede on the pinpointer. If the crew getting pinpointers becomes a big problem then we can lock those some other way, possibly.

**Comment by paprka** [comment]
i really dont think removing the explosive implant is a good idea, it's very out of touch for you to assume theyre only so people don't get their guns (lol? they would have dropped their guns already anyway, since people only active them in crit or when stunned with no hope of getting away)

please reconsider

**Comment by Cheridan** [comment]
>if I don't like a change it means they're out of touch

I KNOW that explosive implants don't effect guns. That's basically the whole point of this. Explosive implants were added for equipment denial but ultimately fail at their role while removing gameplay possibilities.

**Comment by paprka** [comment]
ops being killed by their own gear in 3...2...

**Comment by Lumipharon** [comment]
Was this seriously merged with the explosive implant removal? Op NEED those implants. As soon as one op dies, now their comms are compromised, the crew can track the disk and by extension the nuke/the ops/the shuttle, they have all their gear which almost for certain includes secondary gear such as the pistol and possibly the shotgun, as well as emags, syndie ID's for chasing the nuke with the pinpointer right onto the shuttle etc etc.

This is a terrible idea, and frankly has nothing to do with putting safeties on guns.
